Source/WebKit/UIProcess/Automation/WebDriverBidiProcessor.cpp

```diff
@@ -94,6 +96,77 @@ void WebDriverBidiProcessor::processBidiMessage(const String& message)
     protectedBackendDispatcher()->dispatch(message);
 }
 
+// Translate internal error messages that come from the inspector protocol payload.
+// See the list of allowed bidi errors in https://w3c.github.io/webdriver-bidi/#errors
+static String toBidiErrorCode(int errorCode, const String& inspectorInternalMsg)
+{
+    // These error codes are specified in JSON-RPC 2.0, Section 5.1.
+    switch (errorCode) {
+    case -32700: // Parse error
+    case -32600: // Invalid request
+    case -32603: // Internal error
+    case -32000: // Server error
```

Comment:
All handler errors are server errors, which means that the control flow will never reach parseEnumValueFromString below. You probably need to move the error mapping into this "Server error" branch.

This entire approach to error handling feels overcomplicated, we'll need to address it separately.

Reply:
Nice catch, this is what `CommandResult` does for the classic errors and I missed it while testing.

And I agree the error handling will need some follow-ups changes, as this version is kinda "monkey patching" on top of the classic error handling.